protocols/direct-funding.go

```diff
@@ -103,6 +103,8 @@ func NewDirectFundingObjectiveState(initialState state.State, myAddress types.Ad
 			init.MyDepositSafetyThreshold[assetAddress] = threshold
 			init.MyDepositTarget[assetAddress] = myShare.Add(myShare, threshold)
 		}
+	} else {
+		return init, err
```

Comment:
Here, I would also prefer the "check errors, then do stuff".
```
channelOutcome, err := outcome.Decode(initialState.VariablePart().EncodedOutcome);
if err != nil {
  // PANIC!
}

...
```

Is there a general guideline for golang developers which addresses this style choice? (I understand the verbosity of my snippet above is one of the pain points of golang.)

Reply:
In this case, there's a more concrete benefit of `if [initialize]; condition { [imperative] } else { [handleError] }`: it limits the scope of the initialized variables to just the `if` block.

Maybe it is better for the (much shorter) error condition to occur first: `if [init]; err != nil { handleError() } else { doStuff() }`

All of this aside, I think the entire condition here can / should be scrapped. More on that in another PR.